### PR TITLE
VideoNav Buttons

### LINKED ISSUE #1012: Remote control down arrow should take you to video controls, not subtitle selection
Hi,

I find the client for Roku works okay for me. However, pressing the down arrow on the Roku remote control should bring up the player controls to be able to pause, fast forward, rewind, etc, and not a subtitle selection box. Please change this behavior. A little subtitle selection icon in the player controls to would be sufficient for subtitle selection.

Thanks for this great product. I prefer it to Plex with the exception of this single issue.

### PR DESCRIPTION
**Pull request by candry7731**
Add buttons to the video player. Press down key while watching to get the new nav buttons. 

- [x] need to fix focus issues when dialogs are closed. 

- [x] need to fix cast not populating when button selected

<s>- [ ] add favorite button</s> Will be added in a later PR

- [x] add mini guide button when watching live tv

**Issues**
Guide loads every time the guide button is pressed. 
Guide does not remember the channel number so it always loads at the first channel instead of the current channel posistion

**Fixes**
Fixes #1012

### CODE REVIEW

**Comment by sevenrats** [comment]
- [x] video begins playing when you press back from persondetails. slider is still up.  slider should go down or video should remain paused.
- [x] click a persondetails, press back twice and the video closes.  the slider should close.

**Comment by sevenrats** [comment]
- [x] closing the description dialog by any method makes it impossible to bring the buttons back up
- [x] closing the playback info dialog by any method makes it impossible to bring the buttons back up
- [x] dot operator on invalid line 57 videoplayer.brs
- [x] down and back buttons should get rid of the buttongroup without stopping or closing video (just up feels claustrophobic)
- [ ] ~~buttongroup should still be up when we press down from extras slider, maybe/maybe not when pressing back from dialogs~~

**Comment by 1hitsong** [comment]
- [x] Press down to show buttons. I expected pressing "Back" would close the buttons, but instead if left the video player.
- [x] Press down to show buttons. Press "OK" to select info. Press "OK" to close info. Pressing down doesn't show the buttons.
- [ ] The Info dialog needs a title to go along with the info icon
- [x] Clicking CC when video has no subtitles does nothing. We need to hide the button, disable the button (gray out button, put cross out over it?), or show a dialog telling the user there are no subtitles available.
- [x] Clicking the Cast button when there is no cast data brings up an empty slider. We should do whatever we do with subtitles here as well.
- [ ] Play video. Press down, and choose Stream Info. Press "OK" once to close. With video still paused, press left to rewind. Pressing right doesn't fast forward.
- [ ] Icon text says Stream In.... last two letters are cut off on my Roku TV.
- [ ] Icon text says Stream In..., but first header text I see in the popup says Playback Info.

File: components/music/ArtistView.brs [hunk 1]
```diff
@@ -262,6 +262,7 @@ function onKeyEvent(key as string, press as boolean) as boolean
             if m.top.selectedButtonIndex > 0
                 m.previouslySelectedButtonIndex = m.top.selectedButtonIndex
                 m.top.selectedButtonIndex = m.top.selectedButtonIndex - 1
+                print "index= "m.top.selectedButtonIndex
```
Comment (1hitsong) [inline components/music/ArtistView.brs]: Debug code

File: components/music/ArtistView.brs [hunk 2]
```diff
@@ -270,6 +271,7 @@ function onKeyEvent(key as string, press as boolean) as boolean
                 selectedButton.focus = false
 
                 m.sectionNavigation.setFocus(true)
+                print "if press"
```
Comment (1hitsong) [inline components/music/ArtistView.brs]: Debug code

File: components/music/ArtistView.brs [hunk 3]
```diff
@@ -282,11 +284,12 @@ function onKeyEvent(key as string, press as boolean) as boolean
             if m.buttonGrp.getChild(m.top.selectedButtonIndex).escape = "right"
                 m.buttonGrp.getChild(m.top.selectedButtonIndex).escape = ""
                 m.previouslySelectedButtonIndex = m.top.selectedButtonIndex
-
+                print "first if"
```
Comment (1hitsong) [inline components/music/ArtistView.brs]: Debug code

File: components/music/ArtistView.brs [hunk 4]
```diff
@@ -282,11 +284,12 @@ function onKeyEvent(key as string, press as boolean) as boolean
...
                 if m.top.selectedButtonIndex < m.buttonCount - 1
                     m.top.selectedButtonIndex = m.top.selectedButtonIndex + 1
+                    print "second if"
```
Comment (1hitsong) [inline components/music/ArtistView.brs]: Debug code

File: components/music/ArtistView.brs [hunk 5]
```diff
@@ -282,11 +284,12 @@ function onKeyEvent(key as string, press as boolean) as boolean
...
                 end if
-
+                print "index= "m.top.selectedButtonIndex
```
Comment (1hitsong) [inline components/music/ArtistView.brs]: Debug code

**Review by 1hitsong — CHANGES_REQUESTED** [review]

**Comment by sevenrats** [comment]
I have a pr that just needs to get polished up that essentially fixes all the focus issues.  I will submit it to this branch tomorrow probably. @candry7731 , don't fix any of the above until you see my code unless you're down to duplicate work.

File: components/PersonDetails.brs [hunk 6]
```diff
@@ -5,9 +5,9 @@ sub init()
     m.btnGrp.observeField("escape", "onButtonGroupEscaped")
     m.favBtn = m.top.findNode("favorite-button")
     m.extrasGrp = m.top.findNode("extrasGrp")
-    m.top.findNode("VertSlider").keyValue = "[[30, 998], [30, 789], [30, 580], [30,371 ], [30, 162]]"
-    m.extrasGrp.opacity = 1.0
-    m.extrasGrp.translation = "[30, 998]"
+    'm.top.findNode("VertSlider").keyValue = "[[30, 998], [30, 789], [30, 580], [30,371 ], [30, 162]]"
```
Comment (1hitsong) [inline components/PersonDetails.brs]: Remove commented out code

File: components/PersonDetails.brs [hunk 7]
```diff
@@ -5,9 +5,9 @@ sub init()
...
+    m.extrasGrp.opacity = 0
+    'm.extrasGrp.translation = "[30, 998]"
```
Comment (1hitsong) [inline components/PersonDetails.brs]: Remove commented out code

File: components/JFVideo.brs [hunk 8]
```diff
@@ -212,48 +278,211 @@ sub bufferCheck(msg)
...
-        ' return false to allow selection during seeking
-        if m.top.state = "paused"
-            m.top.control = "resume"
-            return false
-        else if m.top.state = "playing"
-            m.top.control = "pause"
-            return false
+    end if
+
+    if m.buttonGrp.visible = true
```
Comment (1hitsong) [inline components/JFVideo.brs]:
```suggestion
    if m.buttonGrp.visible
```

**Review by 1hitsong — CHANGES_REQUESTED** [review]

**Comment by candry7731** [comment]
@sevenrats, Thanks I will wait for your pr.

**Comment by sevenrats** [comment]
since i had to implement another special-case dialog, i think it would be wise if this PR also renamed the relatively new "PlaybackDialog" to be called "PlaybackFailureDialog" to make it more reasonably distinct from the new "PlaybackInfoDialog"

Other than that, I think this is ready for prime time.

**Comment by 1hitsong** [comment]
- [ ] Switching channels from mini guide didn't work. Resulted in never ending spinner
- [ ] Regression: Pressing "OK" while video plays no longer toggles playback
- [ ] The Info & stream info dialogs need a title to go along with the icon
- [ ] Play video. Press down, and choose Stream Info. Press "OK" once to close. Pressing left will rewind. Pressing right doesn't fast forward.
- [ ] Icon text says "Stream In" last two letters are cut off on my Roku TV.

File: components/JFVideo.brs [hunk 9]
```diff
@@ -141,8 +189,37 @@ sub onState(msg)
...
+                m.buttonGrp.removeChild(m.top.findNode("guide"))
+                setupButtons()
             end if
         end if
+        'check if video is Live TV Channel
+        if m.top.content.live = true
+            m.buttonGrp.removeChild(m.top.findNode("cast"))
+            m.buttonGrp.removeChild(m.top.findNode("cc"))
+            m.getItemQueryTask.live = "true"
+            print "m.top.id" m.top.id
```
Comment (1hitsong) [inline components/JFVideo.brs]: Debug code

**Review by 1hitsong — CHANGES_REQUESTED** [review]

**Comment by candry7731** [comment]
Hmm,

Not sure why any of this would change?? We have tested this on unstable a
ton.....

Ill check it out again.

**Review by candry7731 — COMMENTED** [review]
@1hitsong can you remove this commit? I committed it by accident.

**Comment by candry7731** [comment]
Switching channels from mini guide didn't work. Resulted in never ending spinner
~Regression: Pressing "OK" while video plays no longer toggles playback~
~The Info & stream info dialogs need a title to go along with the icon - DONE~
~Icon text says "Stream In" last two letters are cut off on my Roku TV. - DONE~

~**Still needs work**~
 ~Play video. Press down, and choose Stream Info. Press "OK" once to close. Pressing left will rewind. Pressing right doesn't fast forward.~
~Regression: Pressing "OK" while video plays no longer toggles playback~

BOTH of these issues should now be fixed.

**Review by candry7731 — COMMENTED** [review]
Fixes Regression: Pressing "OK" while video plays no longer toggles playback

File: locale/en_US/translations.ts [hunk 10]
```diff
@@ -803,6 +803,10 @@
         <message>
             <source>Stream Information</source>
             <translation>Stream Information</translation>
+        </message>
+         <message>
```
Comment (cewert) [inline locale/en_US/translations.ts:807]:
```suggestion
        <message>
```

File: locale/en_US/translations.ts [hunk 11]
```diff
@@ -803,6 +803,10 @@
...
+            <source>Program Information</source>
+            <translation>Stream Information</translation>
```
Comment (cewert) [inline locale/en_US/translations.ts:809]:
```suggestion
            <translation>Program Information</translation>
```

File: locale/en_US/translations.ts [hunk 12]
```diff
@@ -942,6 +946,26 @@
             <source>Resumable</source>
             <translation>Resumable</translation>
         </message>
+        <message>
+            <source>Guide</source>
+            <translation>Guide</translation>
+            <extracomment>Trickplaybar button, Guide</extracomment>
```
Comment (cewert) [inline locale/en_US/translations.ts:952]:
```suggestion
            <extracomment>Trickplaybar button text</extracomment>
```

No need to add "guide". It's obvious from context

File: locale/en_US/translations.ts [hunk 13]
```diff
@@ -942,6 +946,26 @@
...
+        </message>
+        <message>
+            <source>Info</source>
+            <translation>Info</translation>
+            <extracomment>Trickplaybar button, Info</extracomment>
```
Comment (cewert) [inline locale/en_US/translations.ts:957]:
```suggestion
            <extracomment>Trickplaybar button text</extracomment>
```

File: locale/en_US/translations.ts [hunk 14]
```diff
@@ -942,6 +946,26 @@
...
+        </message>
+        <message>
+            <source>Cast</source>
+            <translation>Cast</translation>
+            <extracomment>Trickplaybar button, Cast</extracomment>
```
Comment (cewert) [inline locale/en_US/translations.ts:962]:
```suggestion
            <extracomment>Trickplaybar button text</extracomment>
```

File: locale/en_US/translations.ts [hunk 15]
```diff
@@ -942,6 +946,26 @@
...
+        </message>
+        <message>
+            <source>Loop</source>
+            <translation>Loop</translation>
+            <extracomment>Trickplaybar button, Loop</extracomment>
```
Comment (cewert) [inline locale/en_US/translations.ts:967]:
```suggestion
            <extracomment>Trickplaybar button text</extracomment>
```

**Pull request closed without merging**